code/datums/components/team_monitor.dm

```diff
@@ -0,0 +1,535 @@
+	var/hud_on = hud_visible
+	var/mob/user = updating
+	//Remove tracking from old frequency
+	if(team_frequency)
+		if(updating)
+			toggle_hud(FALSE, updating)
+		//Remove from the global frequency
+		GLOB.tracker_huds[team_frequency] -= src
+		//Clear tracking
+		tracking.Cut()
+	team_frequency = "[team_freq_key][new_frequency]"
+	//Add tracking to new frequency
+	if(!team_frequency)
+		return
+	//Adds our tracking component to the global list of trackers
+	add_tracker_hud(team_frequency, src)
+	//Gets the other trackers on our frequency
+	get_matching_beacons()
+	//Show hud if needed
+	if(user)
+		toggle_hud(hud_on, user)
+
+//Adds a new atom to the tracking monitor, will create a hud element that tracks them
+//TODO: Add the screen if already equipped
+//Should be the only way atoms are added to the tracking list
+/datum/component/team_monitor/proc/add_to_tracking_network(datum/component/tracking_beacon/beacon)
+	if(beacon != attached_beacon)
+		if(updating?.hud_used)
+			var/atom/movable/screen/arrow/arrow = new
+			arrow.alpha = 240
+			arrow.color = beacon.colour
+			arrow.hud = updating.hud_used
+			updating.hud_used.team_finder_arrows += arrow
+			tracking[beacon] = arrow
+			//Update arrow direction
+			update_atom_dir(beacon)
+			//Update their hud
+			updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
+		else
+			tracking[beacon] = null
+
+//==================
+// Component
+//  - TRACKER COMPONENT
+//  - HANDLES UPDATING TRACKERS WHEN MOVED
+//==================
+
+/datum/component/tracking_beacon
+	//The frequency of the team signals we are trackings
+	var/team_freq_key = "debug"	//Key <-- cannot be changed
+	var/team_frequency = ""	//Final compiled: Consists of key then numbers between 1 and 999
+	//Are we visible to other trackers?
+	var/visible = TRUE
+	//Our colour
+	var/colour = "#FFFFFF"
+	//Who are we updating for
+	var/mob/updating = null
+	//Do we have an attached monitor?
+	var/datum/component/team_monitor/attached_monitor
+	//Should we update when not equipped?
+	var/always_update = FALSE
+	//Global signal?
+	var/global_signal = FALSE
+
+/datum/component/tracking_beacon/Initialize(_frequency_key, _frequency, _attached_monitor, _visible = TRUE, _colour = "#ffffff", _global = FALSE, _always_update = FALSE)
+	. = ..()
+
+	//Set vars
+	colour = _colour
+	attached_monitor = _attached_monitor
+	always_update = _always_update
+	global_signal = _global
+
+	//Set the frequency we are transmitting on
+	team_freq_key = _frequency_key
+	if(_global)
+		team_frequency = "[_frequency_key]-GLOB"
+	else if(_frequency)
+		team_frequency = "[_frequency_key][_frequency]"
+	else
+		team_frequency = null
+
+	//Add ourselves to the tracking network
+	add_tracker_beacon(team_frequency, src)
+
+	//Register tracking signal
+	if(always_update)
+		RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/update_position)
+	else
+		//Reigster equipping signals
+		RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/parent_equipped)
```

Comment:
I think you might be able to utilize `COMSIG_ITEM_PICKUP`.

Reply:
Equip is better since it also handles putting items on. In this case its about hardsuits being equipped since while a hardsuit isn't worn the beacon is disabled.

Reply:
I was thinking about `always_update` ones, but it is your call.